storage/innobase/srv/srv0start.cc

```diff
@@ -2689,6 +2691,20 @@ dberr_t srv_start(bool create_new_db, const std::string &scan_directories) {
 
       log_sys_close();
 
+      // We have set mysql's dd flags based on what we read from page 0. This
+      // was before redo log was applied. Now we have applied changes from redo
+      // log. If there were changes in redo for page 0 of mysql.ibd we check if
+      // encryption flag has changed. If it did we amend mysql's dd encryption
+      // flags accordingly.
+      if (mysql_ibd_page_0_present_in_redo) {
+        fil_space_t *mysql_space = fil_space_get(dict_sys_t::s_space_id);
+        if (mysql_space == nullptr ||
```

Comment:
it the tablespace is opened by redo, it should not be nullptr.
Can you add ut_ad(mysql_space != nullptr)?